mlir/test/xmir/e2e/bert-torch-tosa/bert_part_0.torch-tosa.mlir

```diff
@@ -0,0 +1,15 @@
+// MLIR#762: Trailing reshape bug
+// RUN: rocmlir-driver -host-pipeline partition -targets %arch %s | FileCheck %s
+// RUN-DISABLE: rocmlir-driver -host-pipeline partition,highlevel -targets %arch %s | rocmlir-gen -ph -print-results -rand 1 -rand_type float -fut bert_part_0 --verifier clone - | rocmlir-driver -host-pipeline xmodel -kernel-pipeline full | xmir-runner --shared-libs=%linalg_test_lib_dir/libmlir_rocm_runtime%shlibext,%conv_validation_wrapper_library_dir/libconv-validation-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_c_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_async_runtime%shlibext --entry-point-result=void | FileCheck %s
+// CHECK-DISABLED: RMS = {{.*}}e-08
+// COM: CHECK: [1 1 1]
+// CHECK: bert_part_0__part_0
+module {
+  func.func @bert_part_0(%arg0: tensor<1x12x384xf32> {func.read_access}, %arg1: tensor<384x384xf32> {func.read_access}, %arg2: tensor<1x1x384xf32> {func.read_access}) -> (tensor<1x12x12x32xf32> {func.write_access}) {
+      %0 = "tosa.reshape"(%arg1) {new_shape = [1, 384, 384]} : (tensor<384x384xf32>) -> tensor<1x384x384xf32>
+      %1 = "tosa.matmul"(%arg0, %0) : (tensor<1x12x384xf32>, tensor<1x384x384xf32>) -> tensor<1x12x384xf32>
+      %2 = "tosa.add"(%1, %arg2) : (tensor<1x12x384xf32>, tensor<1x1x384xf32>) -> tensor<1x12x384xf32>
+      %3 = "tosa.reshape"(%2) {new_shape = [1, 12, 12, 32]} : (tensor<1x12x384xf32>) -> tensor<1x12x12x32xf32>
```

Comment:
Can you make another narrower test of this case just for FoldTranspose pass ?

I think it is the most important functionality this change enables and I think we would to assert this transform happens within the pass itself.

Reply:
This one does not work yet (see DISABLE above). See [762](https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/762).

This would be the case where we couldn't align the linalg.generic out with the in from the matmul.. Will work on that one next.

Reply:
Yeah, I'm on team "narrower tests" here. `--mlir-print-ir-after-all` may be helpful

Reply:
Or at least having a separate test for the bug over in `Dialect/Rock`